### PR TITLE
EOS-16347 - ConfStore: Implement CLI

### PR DESCRIPTION
**Pull request by suryakumar1024**
Signed-off-by: surya <k.suryakumar1024@gmail.com>

py-utils/src/utils/conf_store/conf_cli.py has following code changes

### CODE REVIEW

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import sys
```
Comment (sachinpunadikar) [inline py-utils/src/utils/conf_store/conf_cli.py:1]: Add copyright text here
Also add below as the first line  
#!/usr/bin/python3

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+from cortx.utils.conf_store import Conf
+from cortx.utils.conf_store.error import ConfStoreError
+
+parser = argparse.ArgumentParser(description='ConfStore CLI')
+
+parser.add_argument("url", help="Location of the config file")
+parser.add_argument("command", help="Command could be get()/ set()/ delete()",
+                    choices=['get', 'set', 'delete'], type=str)
+parser.add_argument("key_value", help="key_value could be a list of single or"
```
Comment (sachinpunadikar) [inline py-utils/src/utils/conf_store/conf_cli.py:13]: its always not a key-value pair .. for get and delete its just key or list of keys, only for set it is kay-value pair(s)

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
work on the comments

File: README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ The cortx-utils repository contains utility tools and code for various CORTX sub
 
 ## Contribute to CORTX-Utils
 
-We are excited about your interest in CORTX and hope you will join us. We take our community very seriously, and we are committed to creating a community built on respectful interactions and inclusivity, as documented in our [Code of conduct](CODE_OF_CONDUCT.md).
+We are excited about your interest in CORTX and hope you will join us. We take our community very seriously, and we are committed to creating a community built on respectful interactions and inclusivity, as documented in our [Code of conduct](https://github.com/Seagate/cortx/blob/main/CONTRIBUTING.md#code-of-conduct).
```
Comment (ujjwalpl) [inline README.md:10]: Where is the code to install CLI or include it in the rpm ?

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import sys
+import argparse
+import errno
+
+from cortx.utils.conf_store import Conf
+from cortx.utils.conf_store.error import ConfStoreError
+
+parser = argparse.ArgumentParser(description='ConfStore CLI')
+
+parser.add_argument("url", help="Location of the config file")
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_cli.py:10]: parser.add_argument("url", help="URL of the config location")

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 2 above)
Comment (ujjwalpl) [reply py-utils/src/utils/conf_store/conf_cli.py:13]: Correct. We need to call it kv_spec and have the usage printed accordingly. Refer to the spec.

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+parser.add_argument("key_value", help="key_value could be a list of single or"
+                    " nested. single - key1=value1 nested key1>key2=value2 "
+                    "key1=value1;key2=value2")
+
+args = parser.parse_args()
+index = 'conf_cli'
+Conf.load(index, args.url)
+
+command = args.command
+key_val = args.key_value
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_cli.py:22]: kv_spec = args.key_value
kv_list = kv_spec.split(';')

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+key_val_lst = key_val.split(';')
+
+
+if command == "set":
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_cli.py:26]: Better to create a class. 
```
class ConfCli:
    _index = "conf_cli"
    def __init__(self, url):
        self._conf = Conf.load(self._index, url)

    def set(self, kv_list): 
        for kv in kv_list:
            key, val = kv.split('=')
            Conf.set(key, val)

    def get(self, key_list):
        val = []
        for key in key_list:
            val.append(Conf.get(key))
        return val
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    for key, value in key_val_dict.items():
+        Conf.set(index, key, value)
+    Conf.save(index)
+
+elif command == "get":
+    for key in key_val_lst:
+        vals = Conf.get(index, key)
+        print(vals)
+
+elif command == 'get_keys':
```
Comment (ujjwalpl) [inline py-utils/src/utils/conf_store/conf_cli.py:44]: get_keys is not required for now.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 2 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_cli.py]: Updated in current patch

File: py-utils/src/utils/conf_store/conf_cli.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+args = parser.parse_args()
+index = 'conf_cli'
+Conf.load(index, args.url)
+
+command = args.command
+key_val = args.key_value
+key_val_lst = key_val.split(';')
+
+
```
[the remark this answers is not in the document]
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_cli.py]: Updated in current patch

File: README.md (hunk 3 above)
Comment (suryakumar1024) [reply README.md:10]: Now the CLI will be install along with the cortx-py-utils package itself

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 4 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_cli.py]: Updated in current patch

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 5 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_cli.py]: updated in current patch

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 6 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_cli.py]: Handled in different way to fit in a single callable main function so as to link in setup.py

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 7 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_cli.py]: updated in current patch

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 1 above)
Comment (suryakumar1024) [reply py-utils/src/utils/conf_store/conf_cli.py]: Updated in current patch

File: py-utils/src/utils/conf_store/conf_cli/__init__.py [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# CORTX-Py-Utils: CORTX Python common library.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (sachinpunadikar) [inline py-utils/src/utils/conf_store/conf_cli/__init__.py:2]: Please correct the year as 2021

File: py-utils/src/utils/conf_store/conf_cli/__main__.py [hunk 10]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+# CORTX-Py-Utils: CORTX Python common library.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (sachinpunadikar) [inline py-utils/src/utils/conf_store/conf_cli/__main__.py:4]: Update the year here as well

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Minor comments, please work on the same

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/utils/conf_store/conf_cli.py (hunk 6 above)
Comment (ujjwalpl) [reply py-utils/src/utils/conf_store/conf_cli.py]: That is not correct. You can refer to the function with class as well. 
Class makes the code readable and manageable. 
Currently code looks difficult to read and unstructured.

**Comment by ujjwalpl** [comment]
This work is completed via another PR https://github.com/Seagate/cortx-utils/pull/155

**Pull request closed without merging**